msal/application.py

```diff
@@ -106,16 +110,27 @@ def __init__(
         self.client = self._build_client(client_credential, self.authority)
         self.authority_groups = None
 
+    def _extract_x5c_value(self):
```

Comment:
Glad to see improvement here comparing to the previous implementation! Thanks! And now it seems we can go even further to simplify its tests, if we refactor to completely separate this helper from this class, just like how we implemented the existing `decorate_scope(...)`. Let's try something like this:

```python
def extract_certs(pem_content):
    # Parse raw pem content and returns a list of string
    # Usage: headers = {"x5c": extract_certs(open("my_cert.pem").read())}
    ...
    return ...
```

Now its test cases won't couple with creating an app instance, yet give us better test coverage on [the 3 scenarios](https://github.com/AzureAD/microsoft-authentication-library-for-python/pull/71#discussion_r298740352), with less dependency, and also happen to be more concise.

```python
def test_extract_a_tag_less_cert(self):
    pem = "my_cert"
    self.assertEqual(["my_cert"], extract_certs(pem))

def test_extract_a_tag_enclosed_cert(self):
    pem = """
-----BEGIN CERTIFICATE-----
my_cert
-----END CERTIFICATE-----
"""
    self.assertEqual(["my_cert"], extract_certs(pem))

def test_extract_multiple_tag_enclosed_certs(self):
    pem = """
-----BEGIN CERTIFICATE-----
my_cert1
-----END CERTIFICATE-----

-----BEGIN CERTIFICATE-----
my_cert2
-----END CERTIFICATE-----
"""
    self.assertEqual(["my_cert1", "my_cert2"], extract_certs(pem))
```

Now we can remove the 3 `def test_subject_name_issuer_authentication_*(...)` test cases.